packages/account/src/account.ts

```diff
@@ -500,6 +502,45 @@ export class Account extends AbstractAccount {
     return this.sendTransaction(request);
   }
 
+  async getTransactionCost(
+    transactionRequestLike: TransactionRequestLike,
+    { signatureCallback, quantitiesToContract = [] }: TransactionCostParams = {}
+  ): Promise<TransactionCost> {
+    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
+    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
+    const baseAssetId = this.provider.getBaseAssetId();
+
+    // Fund with fake UTXOs to avoid not enough funds error
+    // Getting coin quantities from amounts being transferred
+    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
+    // Combining coin quantities from amounts being transferred and forwarding to contracts
+    const requiredQuantities = mergeQuantities(coinOutputsQuantities, quantitiesToContract);
+    const resources = this.generateFakeResources(
+      mergeQuantities(
+        [...requiredQuantities],
+        [{ assetId: baseAssetId, amount: bn('100000000000000000') }]
+      )
```

Comment:
@petertonysmith94 Maybe we should make it clear here that this quantity in `baseAssetId` is being added to pay for the TX fee at the `dryRun`

Reply:
a14330cf21d98d3c9b5f0dab5f7914b0122fa704